gr-uhd/grc/gen_uhd_usrp_blocks.py

```diff
@@ -177,6 +177,12 @@
         self.${'$'}{id}.set_time_unknown_pps(uhd.time_spec(0))
         ${'%'} elif sync == 'pc_clock':
         self.${'$'}{id}.set_time_now(uhd.time_spec(time.time()), uhd.ALL_MBOARDS)
+        ${'%'} elif sync == 'pc_clock_next_pps':
+        self.${'$'}{id}.set_time_next_pps(uhd.time_spec(int(time.time()) + 1.0))
+        time.sleep(1)
+        ${'%'} elif sync == 'gps_time':
+        self.${'$'}{id}.set_time_next_pps(uhd.time_spec(self.${'$'}{id}.get_mboard_sensor("gps_time").to_int() + 1.0))
```

Comment:
I'm not entirely sure that this will actually operate directly after a PPS edge, but you're right it's what we do in query_gpsdo_sensors. Please add a comment here explaining why this is valid.

Reply:
Comments added in requested section of code